apps/condo/access.js

```diff
@@ -113,18 +129,101 @@ const rules = {
         return false
     },
 
-    // canUpdatePeople: ({ session }: ListAccessArgs) => {
-    //     if (!session) {
-    //         // No session? No people.
-    //         return false;
-    //     } else if (session.data.role?.canEditOtherPeople) {
-    //         // Can update everyone
-    //         return true;
-    //     } else {
-    //         // Can update yourself
-    //         return { id: session.itemId };
-    //     }
-    // },
+    /*
+
+      Billing
+
+    */
+    canReadBillingIntegrations: isSignedIn,
+    canManageBillingIntegrations: userIsAdmin,
+    canReadBillingIntegrationAccessRights: userIsAdmin,
+    canManageBillingIntegrationAccessRights: userIsAdmin,
+    canReadBillingIntegrationOrganizationContexts: ({ authentication: { item: user } }) => {
+        if (!user) return false
+        if (user.isAdmin) return true
+        return {
+            // TODO(pahaz): wait https://github.com/keystonejs/keystone/issues/4829 (no access check!)
+            // OR: [
+            //     { organization: { employees_some: { user: { id: user.id }, role: { canManageIntegrations: true } } } },
+            //     { integration: { accessRights_some: { user: { id: user.id } } } },
+            // ],
+        }
+    },
+    canManageBillingIntegrationOrganizationContexts: async ({ authentication: { item: user }, originalInput, operation, itemId }) => {
+        if (!user) return false
+        if (user.isAdmin) return true
+        if (operation === 'create') {
+            // NOTE: can create only by the organization integration manager
+            if (!originalInput || !originalInput.organization || !originalInput.organization.connect || !originalInput.organization.connect.id) return false
+            const organizationId = originalInput.organization.connect.id
+            return await checkOrganizationPermission(user.id, organizationId, 'canManageIntegrations')
+        } else if (operation === 'update') {
+            // NOTE: can update by the organization integration manager OR the integration account
+            if (!itemId) return false
+            const context = await getById('BillingIntegrationOrganizationContext', itemId)
+            if (!context) return false
+            const organizationId = context.organization
+            const integrationId = context.integration
```

Comment:
Guess its mistake here, u mean const organizationId = context.integration.connect.id

Otherwise u don't need to define this variables

Reply:
no, it a state from the database.